proposals/bag-o-stacks/Explainer.md

```diff
@@ -134,9 +134,9 @@ absheaptype ::= ... | stack | nostack
 
 ### Life-cycle of a coroutine
 
-A coroutine is started using the `stack.new_switch` instruction. This performs the equivalent of a function call – on a new stack resource. In addition to the arguments normally expected in a function call, an additional argument is provided that is a stack reference to the caller code -- the caller is suspended as a result of the `stack.new_switch` instruction.
+A coroutine is allocated in the suspended state using the `stack.new` instruction. The initial `switch` to the newly allocated coroutine performs the equivalent of a function call on the new stack resource. In addition to the arguments provided to the `switch`, an additional argument is provided that is a stack reference to the caller code -- the caller is suspended as a result of the `switch` instruction.
```

Comment:
I remain uncomfortable with this change:
in particular, stack.bind cannot completely replace the stack.new_switch instruction.

For the sake of argument, lets call the recursive group of types P (for protocol).

Then, clearly

stack.switch x |= x in P

but also,

C|- stack.new f |= f expands to func x [] and x in P

and

C|- stack.bind x y |= x in P and y in P

This is because of the common rt param in all the stack types.

This effectively eliminates one of the main use cases of stack.new_switch - of binding free variables to a coroutine function before entry into the coroutine proper.

In addition, stack.bind seems less powerful and less useful than cont.bind; again, because you cannot leave (or enter) the recursion group P.